File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int left = 0, right = height.length - 1;
        while (left<right){
            max = Math.max(max,(right-left)*Math.min(height[left],height[right]));
            if (height[left]<height[right])
                left++;
            else right--;
        }
//        for (int i = 0; i < height.length; i++) {
//            for (int j = i; j < height.length; j++) {
//                max = Math.max(max,(j-i)*Math.min(height[i],height[j]));
//            }
//        }
        return max;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxy = 0
        left,right = 0, len(height) - 1
        while left < right:
            maxy=max(maxy,min(height[left],height[right])*(right-left))
            if height[left] < height[right]:
                left+=1
            else:
                right-=1
        return maxy
        # return max(min(x[0][1],x[1][1])*(x[1][0]-x[0][0]) for x in list(combinations(list(enumerate(height)),2)))
        